script/release-notes/notes.js

```diff
@@ -300,33 +303,50 @@ const checkCache = async (name, operation) => {
 
 const getPullRequest = async (number, owner, repo) => {
   const name = `${owner}-${repo}-pull-${number}`
+
   return checkCache(name, async () => {
-    try {
-      return await octokit.pulls.get({ number, owner, repo })
-    } catch (error) {
-      // Silently eat 404s.
-      // We can get a bad pull number if someone manually lists
-      // an issue number in PR number notation, e.g. 'fix: foo (#123)'
-      if (error.code !== 404) {
-        throw error
-      }
+    let failCount = 0
```

Comment:
@codebytere I could be missing something here, but I don't think this retry is going to work the way you are expecting because there is no await happening in the async function.

Also, setTimeout is not async friendly unless you wrap it, eg one of the solutions here: https://stackoverflow.com/questions/33289726/combination-of-async-function-await-settimeout